azext_iot/monitor/parsers/issue.py

```diff
@@ -0,0 +1,234 @@
+            logger.info(to_log)
+
+        if self.severity == Severity.warning:
+            logger.warn(to_log)
+
+        if self.severity == Severity.error:
+            logger.error(to_log)
+
+
+class CentralIssue(Issue):
+    def __init__(self, severity: Severity, message: str, device_id="", template_id=""):
+        super(CentralIssue, self).__init__(severity, message, device_id)
+        self.template_id = template_id
+
+        if not self.template_id:
+            self.template_id = "Unknown"
+
+    def log(self):
+        to_log = "[{}] [DeviceId: {}] [TemplateId: {}] {}".format(
+            self.severity.name.upper(), self.device_id, self.template_id, self.message
+        )
+
+        self._log(to_log)
+
+
+class IssueHandler:
+    def __init__(self):
+        self._issues = []
+
+    def add_issue(self, severity: Severity, message: str, device_id=""):
+        self._issues.append(Issue(severity, message, device_id))
+
+    def add_central_issue(
+        self, severity: Severity, message: str, device_id="", template_id=""
+    ):
+        self._issues.append(CentralIssue(severity, message, device_id, template_id))
+
+    def get_all_issues(self) -> List[Issue]:
+        return self._issues
+
+    def get_issues_with_severity(self, severity: Severity) -> List[Issue]:
+        """
+        arguments:
+            severity: Severity
+        returns:
+            all issues where severity equal specified severity
+
+        example:
+            issue_handler.get_issues_with_severity(Severity.info)
+            returns all issues classified as "info"
+        """
+        return [issue for issue in self._issues if issue.severity == severity]
+
+    def get_issues_with_minimum_severity(self, severity: Severity) -> List[Issue]:
+        """
+        arguments:
+            severity: Severity
+        returns:
+            all issues where severity >= specified severity
+
+        example:
+            issue_handler.get_issues_with_minimum_severity(Severity.warning)
+            returns all issues classified as "warning" and "error"
+            "info" will not be included
+        """
+        return [issue for issue in self._issues if issue.severity >= severity]
+
+    def get_issues_with_maximum_severity(self, severity: Severity) -> List[Issue]:
+        """
+        arguments:
+            severity: Severity
+        returns:
+            all issues where severity <= specified severity
+
+        example:
+            issue_handler.get_issues_with_maximum_severity(Severity.warning)
+            returns all issues classified as "warning" and "info"
+            "error" will not be included
+        """
+        return [issue for issue in self._issues if issue.severity <= severity]
+
+
+SUPPORTED_ENCODING = ["utf-8"]
+SUPPORTED_FIELD_NAME_CHARS = ["a-z", "A-Z", "0-9", "_"]
+SUPPORTED_CONTENT_TYPE = ["application/json"]
+SUPPORTED_MESSAGE_HEADERS = []
+
+
+class IssueMessageBuilder:
+    @staticmethod
```

Comment:
We do not want to use staticmethod. Preference is for module level functions instead. Static methods are rare and often from legacy libraries. The father of Python has mentioned in the past that [static methods were an accident](https://mail.python.org/pipermail/python-ideas/2012-May/014969.html). Not using staticmethods is also inline with Azure Python SDK guidelines.